src/server/cluster/cluster_family.h

```diff
@@ -79,27 +86,36 @@ class ClusterFamily {
   // FlowInfo is used to store state, connection, and all auxiliary data
   // that is needed for correct slots (per shard) data transfer
   struct FlowInfo {
-    facade::Connection* conn = nullptr;
+    std::unique_ptr<RestoreStreamer> streamer;
+    ~FlowInfo();
   };
 
   // Whole slots migration process information
   struct MigrationInfo {
     MigrationInfo() = default;
-    MigrationInfo(std::uint32_t flows_num, std::string ip, uint32_t sync_id, uint16_t port,
-                  std::vector<ClusterConfig::SlotRange> slots)
-        : host_ip(ip),
-          flows(flows_num),
-          slots(slots),
-          sync_id(sync_id),
-          port(port),
-          state(ClusterSlotMigration::State::C_CONNECTING) {
-    }
+    MigrationInfo(std::uint32_t flows_num, std::string ip, uint16_t port,
+                  std::vector<ClusterConfig::SlotRange> slots, Context::ErrHandler err_handler);
+
+    void StartFlow(DbSlice* slice, uint32_t sync_id, journal::Journal* journal, io::Sink* dest);
```

Comment:
The MigrationInfo struct is just holding migration information. I dont think function belongs here as its start the migration process